third_party/move/move-bytecode-verifier/src/type_safety.rs

```diff
@@ -673,7 +662,7 @@ fn verify_instr(
 
         Bytecode::StLoc(idx) => {
             let operand = safe_unwrap!(verifier.stack.pop());
-            if &operand != verifier.local_at(*idx) {
+            if !operand.is_subtype_of(verifier.local_at(*idx)) {
```

Comment:
We should perhaps measure the performance implications of this change? Should such changes also be hidden behind a feature flag?